### PR TITLE
chore(ci): Generate code and docs for dependencies updates

### LINKED ISSUE #3326: chore(ci): Run code gen and doc gen on services API updates
TLDR: Automate https://github.com/cloudquery/cloudquery/pull/3324

cc @bernays 

We can use the same approach as our Helm Charts:
https://github.com/cloudquery/helm-charts/blob/8b1b9b5ece94a92e1ac329594645a4b3318a9c4d/.github/workflows/deps_bump_versions.yml#L10

The linked workflow adds a comment to the PR that triggers another workflow that updates the docs.

That way we keep renovate in charge of handling updates (for example it can rebase PRs with conflicts, auto merge non breaking updates, etc.), but ensure the CI works for those API updates PRs

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes #3326, to be merged after https://github.com/cloudquery/cloudquery/pull/3345.
Based on https://github.com/cloudquery/helm-charts/blob/8b1b9b5ece94a92e1ac329594645a4b3318a9c4d/.github/workflows/deps_bump_versions.yml

### CODE REVIEW

File: .github/workflows/deps_trigger_gen.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          GO_MOD_FILE: ${{ steps.changed-files.outputs.all }}
+        with:
+          github-token: ${{ secrets.GH_CQ_BOT }}
+          script: |
+            const { issue: { number: issue_number }, repo: { owner, repo } } = context;
+            const { SHA: sha, GO_MOD_FILE: goModFile } = process.env;
+            // e.g. plugins/source/aws/go.mod
+            const plugin = goModFile.split('/')[2]
+            console.log({ issue_number, owner, repo, sha, plugin });
+            await github.rest.issues.createComment({
```
Comment (erezrokah) [inline .github/workflows/deps_trigger_gen.yml:33]: This comment triggers the `/gen` command for this plugin (to re-use that workflow)

**Comment by yevgenypats** [comment]
Can we just rigger this the same way as we do with docs and linters and see if anything changed ?

**Comment by bbernays** [comment]
> Can we just rigger this the same way as we do with docs and linters and see if anything changed ?

If we do that, then as soon as package update is released then the next pr will have to update (even if it is for a completely different resource)

**Comment by erezrokah** [comment]
> Can we just rigger this the same way as we do with docs and linters and see if anything changed ?

I think that would make our workflows a bit complex, see just AWS https://github.com/cloudquery/cloudquery/compare/main...erezrokah:chore/update_code_docs_prs?expand=1

And this is also missing the checking the branch name logic

**Review by hermanschaaf — APPROVED** [review]
Happy to try it

**Pull request merged**